docs/API_DESIGN.md

```diff
@@ -0,0 +1,690 @@
+    - [`DidWeb`](#didweb)
+      - [Example: Instantiate an existing `did:web`](#example-instantiate-an-existing-didweb)
+      - [Example: Resolve a `did:web`](#example-resolve-a-didweb)
+    - [`DidDht`](#diddht)
+      - [Example: Create \& publish a `did:dht`](#example-create--publish-a-diddht)
+      - [Example: Create a `did:dht`, add to the Core Properties \& publish](#example-create-a-diddht-add-to-the-core-properties--publish)
+      - [Example: Instantiate an existing `did:dht`](#example-instantiate-an-existing-diddht)
+      - [Example: Update a `did:dht`](#example-update-a-diddht)
+      - [Example: Resolve a `did:dht`](#example-resolve-a-diddht)
+  - [`BearerDid`](#bearerdid)
+- [Verifiable Credentials (VCs)](#verifiable-credentials-vcs)
+  - [Data Model 1.1](#data-model-11)
+    - [`NamedIssuer`](#namedissuer)
+    - [`VerifiableCredential`](#verifiablecredential)
+      - [Example: Create a VC \& sign](#example-create-a-vc--sign)
+      - [Example: Verify a VC-JWT](#example-verify-a-vc-jwt)
+- [Presentation Exchange (PEX)](#presentation-exchange-pex)
+  - [`PresentationDefinition`](#presentationdefinition)
+  - [`InputDescriptor`](#inputdescriptor)
+  - [`Constraints`](#constraints)
+  - [`Field`](#field)
+  - [`Optionality`](#optionality)
+  - [`Filter`](#filter)
+
+> [!NOTE]
+> Refer to the [Custom DSL](./CUSTOM_DSL.md) for below syntax definitions.
+
+# Crypto
+
+## `Jwk`
+
+> [!NOTE]
+> Public & private *key material* are currently strictly represented as [Jwk](#jwk-object-oriented-class), but as the requirement for additional representations (ex: CBOR) present themselves, key material will need to be disintermediated via a polymorphic base class such as `PublicKeyMaterial` (which would expose an instance method for `get_verifier_bytes()`) and `PrivateKeyMaterial` (which would expose instance methods for `to_public_key_material()` and `get_signer_bytes()`), both of which would implement `as_jwk()`, `as_cbor()` and any other concrete representations as instance methods.
+
+```pseudocode!
+/// Partial representation of a [JSON Web Key as per RFC7517](https://tools.ietf.org/html/rfc7517).
+/// Note that this is a subset of the spec.
+CLASS Jwk
+  /// Identifies the algorithm intended for use with the key.
+  PUBLIC DATA alg: string
+
+  /// Represents the key type. e.g. EC for elliptic curve, OKP for Edwards curve
+  PUBLIC DATA kty: string
+
+  /// curve name for Elliptic Curve (EC) and Edwards Curve (Ed) keys.
+  /// e.g. secp256k1, Ed25519
+  PUBLIC DATA crv: string
+
+  /// X coordinate for EC keys, or the public key for OKP.
+  PUBLIC DATA x: string
+
+  /// Y coordinate for EC keys.
+  PUBLIC DATA y: string?
+
+  /// Private key component for EC or OKP keys.
+  PUBLIC DATA d: string?
+```
+
+## Key Managers 
+
+### `KeyManager`
+
+```pseudocode!
+INTERFACE KeyManager
+  /// Returns the signer for the given public key.
+  METHOD get_signer(public_jwk: Jwk): Signer
+```
+
+### `InMemoryKeyManager`
```

Comment:
can we clarify that this is to be used for test purposes only

Reply:
done